appeals/web/src/server/appeals/appeal-details/appeal-details.controller.js

```diff
@@ -25,48 +17,23 @@ export const viewAppealDetails = async (request, response) => {
 		.getAppealDetailsFromId(request.apiClient, request.params.appealId)
 		.catch((error) => logger.error(error));
 
+	const session = request.session;
+
 	if (appealDetails) {
-		const mappedAppealDetailsSummary =
-			mapAppealDetailsToAppealDetailsSummaryParameters(appealDetails);
-		const mappedAppealDetailsSummaryListBuilderParameters =
-			mapAppealDetailsToSummaryListBuilderParameters(appealDetails);
-		const mappedAppealDetailsTableBuilderParameters =
-			mapAppealDetailsToTableBuilderParameters(appealDetails);
-		const appealReferenceFragments = appealDetails?.appealReference.split('/');
-		const mappedNotificationBannerParameters = mapSessionDataToNotificationBannerParameters(
-			request.session
-		);
+		const currentUrl = request.protocol + '://' + request.get('host') + request.originalUrl;
+
+		const pageComponents = appealsDetailPage({ appeal: appealDetails }, currentUrl, session);
+		const shortAppealReference =
+			appealDetails?.appealReference.split('/')[
+				appealDetails?.appealReference.split('/').length - 1
+			];
```

Comment:
Perhaps the `appealShortReference` function (`appeals/web/src/server/lib/nunjucks-filters/appeals.js`) could be used here rather than duplicating the functionality?

Reply:
👍